public/app/features/plugins/extensions/usePluginLinks.tsx

```diff
@@ -0,0 +1,98 @@
+import { useMemo } from 'react';
+import { useObservable } from 'react-use';
+import { isString } from 'util';
+
+import { PluginExtensionLink, PluginExtensionTypes } from '@grafana/data';
+import {
+  UsePluginLinksOptions,
+  UsePluginLinksResult,
+} from '@grafana/runtime/src/services/pluginExtensions/getPluginExtensions';
+
+import { AddedLinksRegistry } from './registry/AddedLinksRegistry';
+import {
+  generateExtensionId,
+  getLinkExtensionOnClick,
+  getLinkExtensionOverrides,
+  getLinkExtensionPathWithTracking,
+  getReadOnlyProxy,
+  logWarning,
+} from './utils';
+
+// Returns an array of component extensions for the given extension point
+export function createUsePluginLinks(registry: AddedLinksRegistry) {
+  const observableRegistry = registry.asObservable();
+
+  return function usePluginLinks({
+    limitPerPlugin,
+    extensionPointId,
+    context,
+  }: UsePluginLinksOptions): UsePluginLinksResult {
+    const registry = useObservable(observableRegistry);
+
+    return useMemo(() => {
+      if (!registry || !registry[extensionPointId]) {
+        return {
+          isLoading: false,
+          links: [],
+        };
+      }
+      const frozenContext = context ? getReadOnlyProxy(context) : {};
+      // We don't return the extensions separated by type, because in that case it would be much harder to define a sort-order for them.
```

Comment:
Hmm, does this comment still make sense?

Reply:
No you're right 👍